### ISSUE TITLE
Using Prefix: Friendly Name & Entity ID don't match

### ISSUE THREAD

**Issue by dsellers1** [sleevezipper/hass-workstation-service#131]
In HassWS, I set a prefix of "desktop_". Entity_IDs show correctly but the friendly name has a hyphen (-) between the prefix and the sensor name.

For example, in Home Assistant I see:
Name = desktop_-CPULoad
Entity ID = sensor.desktop_cpuload
![J7vokkp 1](https://user-images.githubusercontent.com/67642332/141945819-fb0d6462-c935-4ca2-9264-dd4ce0383a13.png)

**Comment by SecretiveShell** [comment]
This should be an easy fix. I will take a look.

**Comment by SecretiveShell** [comment]
It's because of [this issue](https://www.home-assistant.io/docs/configuration/troubleshooting/#entity-names). Hass.io does not support `-` characters in entity names.

**Comment by dsellers1** [comment]
> It's because of [this issue](https://www.home-assistant.io/docs/configuration/troubleshooting/#entity-names). Hass.io does not support `-` characters in entity names.

That's the problem. The hyphen isn't supposed to be there. HassWS is setting the friendly name to include the hyphen. 

In my example posted above, the prefix was set as "desktop_" (which after some experimentation "desktop" would accomplish the same since it looks like HassWS appends an underscore automatically if there isn't one).

The information from the link is talking about the Entity ID's name, not the friendly name. Entity IDs can only have lowercase letters, numbers, and underscores. The friendly name is more flexible and can include spaces.

Even hovering over the What's This? shows the name with a hyphen and entity ID with an underscore. They don't match. In theory, the name could simply replace the hyphen with a space.
![OlKPlPF 1](https://user-images.githubusercontent.com/67642332/142742308-35844205-d03a-4a46-9483-f5168c383b71.png)

**Comment by SecretiveShell** [comment]
can you please confirm that you are expecting the attributes to be :

entity Name : `desktop ActiveWindowSensor`
entity ID : `sensor.desktop_activewindow`

**Comment by dsellers1** [comment]
> 

correct. bonus points if you can make the first letter of the entity name capitalized.

entity Name : Desktop ActiveWindowSensor
entity ID : sensor.desktop_activewindowsensor
